Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/InputActionsEditorWindow.cs

```diff
@@ -210,6 +210,11 @@ private void DirtyInputActionsEditorWindow(InputActionsEditorState newState)
         private bool HasAssetChanged(SerializedObject serializedAsset)
         {
             var asset = (InputActionAsset)serializedAsset.targetObject;
+
+            // Checks if the new asset is empty. If it is, there's nothing to save.
+            if (asset.actionMaps.Count == 0 && asset.controlSchemes.Count == 0)
+                return false;
+
             var newAssetJson = asset.ToJson();
             return newAssetJson != m_AssetJson;
```

Comment:
I would recommend determining (if not really checked) why the this evaluation evaluates to true instead of false (?) instead of introducing additional conditions for some corner case. Isn't the asset json for the newly created asset "empty" similar to what asset.ToJson() would produce or is there any difference there to look into and resolve?

Reply:
Currently, the asset on disk has a specific layout.
Ideally, the layout shouldn't be `"{}"`. It should be a JSON serialized version of an empty asset. 
However, the `name` field is the problem. If you create and rename an input asset, its `name` field inside the asset is not updated at that stage. When opening the asset, there are actually changes to be made (the asset is flagged dirty) since the filename was changed.
The reason why this seems to happen is because on importing an asset, [we're actually setting `asset.name` based on the filename.](https://github.com/Unity-Technologies/InputSystem/blob/b0bcfbfbd0aa0561069447e1ec76d2ba7ec042fa/Packages/com.unity.inputsystem/InputSystem/Editor/AssetImporter/InputActionImporter.cs#L85-L87)

I spent some time on this and am feeling it can become a rabbit hole and introduce regressions. But I think we should revisit this workflow.

Reply:
Sounds odd we are setting name on import and not directly after constructing the asset, e.g. create, import, set name, write if needed, but I think you know best what is the best way to resolve this.